activesupport/lib/active_support/notifications/fanout.rb

```diff
@@ -113,11 +116,33 @@ def self.wrap_all(pattern, subscriber)
           end
         end
 
+        class Matcher #:nodoc:
+          attr_reader :pattern, :blacklist
+
+          def self.wrap(pattern)
+            return pattern if String === pattern
+            new(pattern)
+          end
+
+          def initialize(pattern)
+            @pattern = pattern
+            @blacklist = Set.new
```

Comment:
I believe we're trying to avoid `@blacklist`. I think `@unsubscribed` would work well and be as/more descriptive 🎉